### PR TITLE
add ontology address

### PR DESCRIPTION
**Pull request by NashMiao**
## Description

1. add ontology address
2. add unit test for ontology address

## Testing instructions

Run the unit test in `tests/Ontology/AddressTests.cpp`

## Types of changes

New feature (non-breaking change which adds functionality), support ontology address.

## Checklist

- [x] Add tests to cover changes as needed.
- [x] Coin configuration
- - [x] Add coin type / decimals
- - [x] Return correct curve and purpose in `src/Coin.cpp`.
- [x] Address
- - [x] Derivation from private key in `src/Coin.cpp`.
- - [x] Validation in `src/Coin.cpp`.
- [x] A swift Ontology class inherited from Blockchain and tests
- [x] Add a swift address test case
- [x] Add config and test to TWCoinTypeConfiguration.cpp

### CODE REVIEW

File: src/Ontology/Address.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#include "HexCoding.h"
+#include <TrezorCrypto/sha2.h>
+#include <TrezorCrypto/base58.h>
+#include <TrezorCrypto/ripemd160.h>
+#include <iostream>
+#include <stdexcept>
+
+using namespace TW::Ontology;
+
+std::vector<uint8_t> Address::sha256(std::vector<uint8_t> &data) {
```
Comment (hewigovens) [inline src/Ontology/Address.cpp]: Why not use `Hash::sha256`?

File: src/Ontology/Address.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    std::vector<uint8_t> uc_vec(&digest[0], &digest[SHA256_DIGEST_LENGTH]);
+    return uc_vec;
+}
+
+std::vector<uint8_t> Address::hash256(std::vector<uint8_t> &data) {
+    auto digest = Address::sha256(data);
+    return Address::sha256(digest);
+}
+
+std::vector<uint8_t> Address::ripemd160(std::vector<uint8_t> &data) {
```
Comment (hewigovens) [inline src/Ontology/Address.cpp]: try `Hash::ripemd`

File: src/Ontology/Address.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+#pragma once
+
+#include "../PublicKey.h"
+
+#include <array>
+#include <vector>
+
+namespace TW {
+    namespace Ontology {
```
Comment (hewigovens) [inline src/Ontology/Address.h]: please follow existing code styles

File: src/Ontology/Address.cpp (hunk 1 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: sorry, I'm not familiar with the functions in library. using `Hash::sha256` would be more suitable, I will correct it.

File: src/Ontology/Address.cpp (hunk 2 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: I will correct as soon as possible.

File: src/Ontology/Address.h (hunk 3 above)
Comment (NashMiao) [reply src/Ontology/Address.h]: OK, I will correct as soon as possible.

File: src/Ontology/Address.cpp (hunk 1 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: https://github.com/TrustWallet/wallet-core/pull/103/commits/0f8bf1c78da5f7cd69389bca0146a29aea5c9586

File: src/Ontology/Address.cpp (hunk 2 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: https://github.com/TrustWallet/wallet-core/pull/103/commits/276afdf4d14c0d5300a7fedc436a1f21d2ad0515

File: src/Ontology/Address.h (hunk 3 above)
Comment (NashMiao) [reply src/Ontology/Address.h]: https://github.com/TrustWallet/wallet-core/pull/103/commits/4a5c24f1c48099d58ceae0deccc656a6aa3fb8e6

File: src/Ontology/Address.cpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    std::vector<uint8_t> data(size);
+    std::copy(zero.rbegin(), zero.rend(), data.begin());
+    return hex(data);
+}
+
+std::string Address::b58String() const {
+    std::vector<uint8_t> data(size + 1);
+    data[0] = version;
+    std::copy(zero.begin(), zero.end(), data.begin() + 1);
+    auto checksum = Hash::sha256(Hash::sha256(data));
```
Comment (hewigovens) [inline src/Ontology/Address.cpp]: maybe you can use `base58_encode_check`

File: src/Ontology/Address.cpp [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+std::vector<uint8_t> Address::toScriptHash(std::vector<uint8_t> &data) {
+    return Hash::ripemd(Hash::sha256(data));
+}
+
+bool Address::isValid(const std::string &string) {
+    const auto data = parse_hex(string);
+    return Address::isValid(data);
+}
+
+std::string Address::hexString() const {
```
Comment (hewigovens) [inline src/Ontology/Address.cpp]: Do we really need hex string address format? Where to use it?

File: tests/Ontology/AddressTests.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+#include "PrivateKey.h"
+#include "PublicKey.h"
+#include "HexCoding.h"
+
+#include <gtest/gtest.h>
+
+using namespace TW;
+using namespace TW::Ontology;
+
+TEST(OntologyAddress, Invalid) {
```
Comment (hewigovens) [inline tests/Ontology/AddressTests.cpp]:
```suggestion
TEST(OntologyAddress, Validation) {
```

Not all the asserts are false

File: tests/Ontology/AddressTests.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+TEST(OntologyAddress, fromB58Str) {
+    auto b58Str = "AYTxeseHT5khTWhtWX1pFFP1mbQrd4q1zz";
+    EXPECT_EQ(b58Str, Address(b58Str).b58String());
+    auto errB58Str = "AATxeseHT5khTWhtWX1pFFP1mbQrd4q1zz";
+    EXPECT_THROW(new Address(errB58Str), std::runtime_error);
+}
+
+TEST(OntologyAddress, fromVec) {
```
Comment (hewigovens) [inline tests/Ontology/AddressTests.cpp]:
```suggestion
TEST(OntologyAddress, fromData) {
```

File: src/Ontology/Address.cpp (hunk 5 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: The smart contract address is in the form of hex string & The wallet account address is in the form of base58 encode string.

File: tests/Ontology/AddressTests.cpp (hunk 6 above)
Comment (NashMiao) [reply tests/Ontology/AddressTests.cpp]: OK, it's my fault.

File: tests/Ontology/AddressTests.cpp (hunk 7 above)
Comment (NashMiao) [reply tests/Ontology/AddressTests.cpp]: OK.

Comment (NashMiao) [reply tests/Ontology/AddressTests.cpp]: https://github.com/TrustWallet/wallet-core/pull/103/commits/c6674a786a3319a2963a4b4af36ce704d0c47645

File: tests/Ontology/AddressTests.cpp (hunk 6 above)
Comment (NashMiao) [reply tests/Ontology/AddressTests.cpp]: https://github.com/TrustWallet/wallet-core/pull/103/commits/c6674a786a3319a2963a4b4af36ce704d0c47645

**Comment by alejandro-isaza** [comment]
Updated checklist, please complete.

**Comment by NashMiao** [comment]
> Updated checklist, please complete.
> https://github.com/TrustWallet/wallet-core/blob/master/.github/ISSUE_TEMPLATE/new_blockchain.md#checklist

This one is enough? https://github.com/TrustWallet/wallet-core/issues/104

**Comment by NashMiao** [comment]
> Please rebase on latest master

I had fix conflict.

File: include/TrustWalletCore/TWOntologyAddress.h [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+/// Creates an address from a public key.
+TW_EXPORT_STATIC_METHOD
+struct TWOntologyAddress *_Nonnull TWOntologyAddressCreateWithPublicKey(struct TWPublicKey *_Nonnull publicKey);
+
+TW_EXPORT_METHOD
+void TWOntologyAddressDelete(struct TWOntologyAddress *_Nonnull address);
+
+/// Returns the address string representation.
+TW_EXPORT_PROPERTY
+        TWString *_Nonnull TWOntologyAddressDescription(struct TWOntologyAddress *_Nonnull address);
```
Comment (alejandro-isaza) [inline include/TrustWalletCore/TWOntologyAddress.h]: Remove indentation please

File: include/TrustWalletCore/TWOntologyAddress.h [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+/// Returns the key hash.
+TW_EXPORT_PROPERTY
+        TWData *_Nonnull TWOntologyAddressKeyHash(struct TWOntologyAddress *_Nonnull address);
```
Comment (alejandro-isaza) [inline include/TrustWalletCore/TWOntologyAddress.h]: Remove indentation

File: src/Coin.cpp [hunk 10]
```diff
@@ -225,11 +228,8 @@ DerivationPath TW::derivationPath(TWCoinType coin) {
     case TWCoinTypeXDai:
     case TWCoinTypeZcash:
     case TWCoinTypeZcoin:
-        return DerivationPath(purpose(coin), coin, 0, 0, 0);
-
     case TWCoinTypeAion:
-        return DerivationPath(purpose(coin), coin, 0, 0x80000000, 0x80000000);
-
+        return DerivationPath(purpose(coin), coin, 0, 0, 0);
```
Comment (alejandro-isaza) [inline src/Coin.cpp]: This is incorrect, please revert this change.

File: src/Coin.cpp [hunk 11]
```diff
@@ -28,6 +29,9 @@ using namespace TW;
 
 std::string TW::loadAddress(TWCoinType coin, const Data& data) {
     switch (coin) {
+    case TWCoinTypeOntology:
```
Comment (alejandro-isaza) [inline src/Coin.cpp]: Please order alphabetically.

File: src/Ontology/Address.cpp [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+#include <TrezorCrypto/sha2.h>
+#include <TrezorCrypto/base58.h>
+#include <TrezorCrypto/ripemd160.h>
+
+using namespace TW::Ontology;
+
+Address::Address(const PublicKey &publicKey) {
+    std::vector<uint8_t> builder(publicKey.bytes);
+    builder.insert(builder.begin(), 0x21);
+    builder.push_back(0xAC);
```
Comment (alejandro-isaza) [inline src/Ontology/Address.cpp]: Let's define constants for these magic numbers.

File: src/Ontology/Address.cpp [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    auto data = toScriptHash(builder);
+    std::copy(data.begin(), data.end(), zero.begin());
+}
+
+
+Address::Address(const std::string &b58Address) {
+    size_t dataSz = 25;
```
Comment (alejandro-isaza) [inline src/Ontology/Address.cpp]:
```suggestion
    size_t dataSize = 25;
```

File: src/Ontology/Address.cpp [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    std::array<uint8_t, 25> data{};
+    b58tobin(data.data(), &dataSz, b58Address.data());
+    if (data[0] != version) {
+        throw std::runtime_error("Invalid address version.");
+    }
+    std::vector<uint8_t> addressWithVer(size + 1);
```
Comment (alejandro-isaza) [inline src/Ontology/Address.cpp]:
```suggestion
    Data addressWithVersion(size + 1);
```

File: src/Ontology/Address.h [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+private:
+
+    std::vector<uint8_t> toScriptHash(std::vector<uint8_t> &data);
+
+public:
+
+    static const size_t size = 20;
+    static const uint8_t version = 0x17;
+
+    std::array<uint8_t, size> zero;
```
Comment (alejandro-isaza) [inline src/Ontology/Address.h]: zero?

Comment (NashMiao) [reply src/Ontology/Address.h]: right, I follow the var name in Ontology source code, I think it will convenient to other developer to maintain these source code.

File: src/Ontology/Address.cpp (hunk 12 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: This is VM op code, we only need two op code. Besides, the op code's name make no sense in here. You will see more op code in transaction serialization, we will also provide op code in this way. otherwise, I will spend more than a month to handle transaction serialization. It's unacceptable.

File: src/Ontology/Address.h (hunk 15 above)
Comment (alejandro-isaza) [reply src/Ontology/Address.h]: I disagree, this should be called `bytes` or `data`

File: src/Coin.cpp (hunk 11 above)
Comment (NashMiao) [reply src/Coin.cpp]: This is a strict requirement...

File: src/Coin.cpp (hunk 10 above)
Comment (NashMiao) [reply src/Coin.cpp]: revert incorrect change.

File: src/Ontology/Address.h (hunk 15 above)
Comment (NashMiao) [reply src/Ontology/Address.h]: I think `h160` is better. Install of `bytes` or `data `...

Comment (NashMiao) [reply src/Ontology/Address.h]: Anyway, I will change var name to h160. and if you still not satisfied with this name, you can change it, the source code is under you control instead of me.

File: include/TrustWalletCore/TWOntologyAddress.h (hunk 8 above)
Comment (NashMiao) [reply include/TrustWalletCore/TWOntologyAddress.h]: I had changed in newest code, maybe you need to resolve conflicts first. then, my newest code you can see.

File: include/TrustWalletCore/TWOntologyAddress.h (hunk 9 above)
Comment (NashMiao) [reply include/TrustWalletCore/TWOntologyAddress.h]: I had changed in newest code, maybe you need to resolve conflicts first. then, my newest code you can see.

File: src/Coin.cpp (hunk 10 above)
Comment (NashMiao) [reply src/Coin.cpp]: I had changed in newest code, maybe you need to resolve conflicts first. then, my newest code you can see.

File: src/Coin.cpp (hunk 11 above)
Comment (alejandro-isaza) [reply src/Coin.cpp]: Sarcasm is not appropriate for code review comments. If you don't agree with may comments please say so in plain language.

File: src/Ontology/Address.h (hunk 15 above)
Comment (alejandro-isaza) [reply src/Ontology/Address.h]: You can in fact name it whatever you want. But please don't waste my time by posting a PR for which are not willing to receive feedback on.

File: src/Coin.cpp (hunk 11 above)
Comment (NashMiao) [reply src/Coin.cpp]: I'm sorry, let's continue to cooperate.

File: src/Ontology/Address.cpp (hunk 13 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: I had changed in newest code, maybe you need to resolve conflicts first. then, my newest code you can see.

File: src/Ontology/Address.cpp (hunk 14 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: I had changed in newest code, maybe you need to resolve conflicts first. then, my newest code you can see.

File: src/Ontology/Address.h (hunk 15 above)
Comment (NashMiao) [reply src/Ontology/Address.h]: I receive feedback that really make sense, such as @hewigovens provide. In fact, you provide me some feedback that more like a personal preference, such as order alphabetically. however, the source code in your repository not be ordered alphabetically.

File: src/Coin.cpp (hunk 10 above)
Comment (alejandro-isaza) [reply src/Coin.cpp]: I still see an incorrect value, unfortunately I can't approve unless you revert this change.

File: src/Coin.cpp (hunk 11 above)
Comment (alejandro-isaza) [reply src/Coin.cpp]: It is not a strict requirement, just a suggestion.

File: include/TrustWalletCore/TWOntologyAddress.h (hunk 8 above)
Comment (NashMiao) [reply include/TrustWalletCore/TWOntologyAddress.h]: https://github.com/TrustWallet/wallet-core/pull/103/commits/c1a7cf052dbbe12814dd0b4b574a1d7b65e421ae

File: include/TrustWalletCore/TWOntologyAddress.h (hunk 9 above)
Comment (NashMiao) [reply include/TrustWalletCore/TWOntologyAddress.h]: https://github.com/TrustWallet/wallet-core/pull/103/commits/c1a7cf052dbbe12814dd0b4b574a1d7b65e421ae

File: src/Ontology/Address.cpp (hunk 5 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: https://github.com/TrustWallet/wallet-core/pull/103/commits/8eac3fbeb8a318e3a7139983fe6012f9ec3bc28f

File: src/Coin.cpp (hunk 10 above)
Comment (NashMiao) [reply src/Coin.cpp]: `DerivationPath(purpose(coin), coin, 0, 0, 0);` is right, which is same with your repo‘s master branch: https://github.com/TrustWallet/wallet-core/blob/master/src/Coin.cpp#L262

Now, my newest commit also this one: https://github.com/TrustWallet/wallet-core/pull/103/commits/db8130f5fa2068434da20f4919abac1f10b37717

File: src/Coin.cpp (hunk 11 above)
Comment (NashMiao) [reply src/Coin.cpp]: OK, I'm sorry, I will adopt this suggestion in my code.

File: src/Ontology/Address.cpp (hunk 12 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: I add op code's name in this commit: https://github.com/TrustWallet/wallet-core/pull/103/commits/8eac3fbeb8a318e3a7139983fe6012f9ec3bc28f

File: src/Ontology/Address.h (hunk 15 above)
Comment (NashMiao) [reply src/Ontology/Address.h]: https://github.com/TrustWallet/wallet-core/pull/103/commits/31af0e099566ea2fe7863a14d71107472e968c60

File: src/Ontology/Address.cpp (hunk 14 above)
Comment (NashMiao) [reply src/Ontology/Address.cpp]: https://github.com/TrustWallet/wallet-core/pull/103/commits/8eac3fbeb8a318e3a7139983fe6012f9ec3bc28f

**Comment by alejandro-isaza** [comment]
Looks good. But looks like we need to implement nist256p1 first: `/Users/vsts/agent/2.148.0/work/1/s/src/Coin.cpp:347:77: error: no member named 'nist256p1' in 'TW::PublicKeyType'
            return Ontology::Address(privateKey.getPublicKey(PublicKeyType::nist256p1)).b58String();`

**Comment by NashMiao** [comment]
> Looks good. But looks like we need to implement nist256p1 first: `/Users/vsts/agent/2.148.0/work/1/s/src/Coin.cpp:347:77: error: no member named 'nist256p1' in 'TW::PublicKeyType' return Ontology::Address(privateKey.getPublicKey(PublicKeyType::nist256p1)).b58String();`

Right, I implement nist256p1 in local. But nist256p1's public key tag conflict with secp256k1, I can't finish signature verification interface.

File: include/TrustWalletCore/TWOntologyAddress.h [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+#include "TWBase.h"
+#include "TWData.h"
+#include "TWString.h"
+
+TW_EXTERN_C_BEGIN
+
+struct TWPublicKey;
+
+/// Represents an Ethereum address.
```
Comment (hewigovens) [inline include/TrustWalletCore/TWOntologyAddress.h]:
```suggestion
/// Represents an Ontology address.
```

File: src/Coin.cpp [hunk 17]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (hewigovens) [inline src/Coin.cpp]: There is extra line here

**Comment by hewigovens** [comment]
I have merged and fixed swift project build error, please finish these items if you want to merge it:

1. A swift `Ontology` class inherited from `Blockchain` and tests
2. Add a swift address test case
3. Add config and test to `TWCoinTypeConfiguration.cpp`

**Comment by NashMiao** [comment]
> I have merged and fixed swift project build error, please finish these items if you want to merge it:
> 
> 1. A swift `Ontology` class inherited from `Blockchain` and tests
> 2. Add a swift address test case
> 3. Add config and test to `TWCoinTypeConfiguration.cpp`

OK, I will finish all of these as soon as possible.

File: src/PublicKey.h [hunk 18]
```diff
@@ -41,7 +41,7 @@ class PublicKey {
             return PublicKeyType::ed25519;
         case 2:
         case 3:
-            return PublicKeyType::secp256k1;
+            return PublicKeyType::nist256p1;
```
Comment (hewigovens) [inline src/PublicKey.h]: It's safe to change this line? @alejandro-isaza

File: swift/Podfile.lock [hunk 19]
```diff
@@ -13,4 +13,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: bcab653390ccf2f8e377a439713b531df60e7b01
 
-COCOAPODS: 1.6.1
+COCOAPODS: 1.6.0
```
Comment (hewigovens) [inline swift/Podfile.lock]: please update your cocoapods version to 1.6.1

File: swift/Tests/Addresses/OntologyAddressTests.swift [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+        XCTAssertNil(OntologyAddress(string: "dshadghasdghsadadsadjsad"))
+    }
+    
+    func testIsValid() {
+        XCTAssertFalse(OntologyAddress.isValidString(string: "abc"))
+        XCTAssertFalse(OntologyAddress.isValidString(string: "0x5aAeb6053F3E94C9b9A09f33669435E7Ef1BeAed"))
+        XCTAssertTrue(OntologyAddress.isValidString(string: "ANDfjwrUroaVtvBguDtrWKRMyxFwvVwnZD"))
+    }
+    
+    func testFromPublicKey() {
```
Comment (hewigovens) [inline swift/Tests/Addresses/OntologyAddressTests.swift]: please add `testFromPrivateKey` too

File: swift/Tests/HDWalletTests.swift [hunk 21]
```diff
@@ -88,6 +88,15 @@ class HDWalletTests: XCTestCase {
         XCTAssertEqual("hx78c6f744c68d48793cd64716189c181c66907b24", address0.description)
         XCTAssertEqual("hx92373c16531761b31a7124c94718da43db8c9d89", address1.description)
     }
+    
+    func testDeriveOntology() {
+        let blockchain = Ontology()
+        let wallet = HDWallet.test
+        let key = wallet.getKey(at: blockchain.derivationPath(at: 0))
```
Comment (hewigovens) [inline swift/Tests/HDWalletTests.swift]: There are simpler APIs:
```swift
        let key = wallet.getKeyForCoin(coin: .ontology)
        let address = CoinType.ontology.deriveAddress(privateKey: key)
```

File: swift/Tests/HDWalletTests.swift [hunk 22]
```diff
@@ -88,6 +88,15 @@ class HDWalletTests: XCTestCase {
...
+        let address = blockchain.address(for: key.getPublicKeyNist256p1())
+
+        XCTAssertEqual("AH11LGtFk6VU9Z7suuM5eNpho1bAoE5Gbz", address.description)
```
Comment (hewigovens) [inline swift/Tests/HDWalletTests.swift]: Do you cross check with other wallets? we assume same phrases produce same ONT address

File: tests/CoinTests.cpp [hunk 23]
```diff
@@ -57,6 +57,12 @@ TEST(Coin, validateAddressZcash) {
     EXPECT_FALSE(validateAddress(TWCoinTypeZcash, "1Ma2DrB78K7jmAwaomqZNRMCvgQrNjE2QC"));
 }
 
+TEST(Coin, validateAddressOntology){
+    EXPECT_TRUE(validateAddress(TWCoinTypeOntology,"AW2f6d82p7dC3jFMrzR5x5GhyTWSB5pxVz"));
```
Comment (hewigovens) [inline tests/CoinTests.cpp]:
```suggestion
    EXPECT_TRUE(validateAddress(TWCoinTypeOntology, "AW2f6d82p7dC3jFMrzR5x5GhyTWSB5pxVz"));
```

**Comment by hewigovens** [comment]
Do you want to include `Signer` in this PR, it seems not finished yet @NashMiao

**Comment by NashMiao** [comment]
> Do you want to include `Signer` in this PR, it seems not finished yet @NashMiao

I will delete it and provide it in next PR, thanks.

File: src/PublicKey.h (hunk 18 above)
Comment (NashMiao) [reply src/PublicKey.h]: No, this change is for my unit test code. and this problem may need you to help solve. As I mentioned last week, `secp256k1` & `nist256p1` 's public key tag are same. we can't verify signature in `nist256p1` now.

Comment (NashMiao) [reply src/PublicKey.h]: I will revert this change, sorry.

File: swift/Tests/HDWalletTests.swift (hunk 22 above)
Comment (NashMiao) [reply swift/Tests/HDWalletTests.swift]: Right, we do cross check by ontology-ts-sdk, which used by other wallet & chrome wallet plugin.

**Comment by NashMiao** [comment]
I had remove this test file, maybe CI will recovery.

image

File: swift/Tests/HDWalletTests.swift (hunk 21 above)
Comment (NashMiao) [reply swift/Tests/HDWalletTests.swift]: Hi, here: https://github.com/hsiaosiyuan0/wallet-core/commit/219655ceb4cf5ef4ea32c57634ff3b283708f0cd

File: swift/Tests/Addresses/OntologyAddressTests.swift (hunk 20 above)
Comment (NashMiao) [reply swift/Tests/Addresses/OntologyAddressTests.swift]: Hi, here: https://github.com/hsiaosiyuan0/wallet-core/commit/219655ceb4cf5ef4ea32c57634ff3b283708f0cd

File: swift/Podfile.lock (hunk 19 above)
Comment (NashMiao) [reply swift/Podfile.lock]: Hi, here: https://github.com/hsiaosiyuan0/wallet-core/commit/219655ceb4cf5ef4ea32c57634ff3b283708f0cd

**Comment by NashMiao** [comment]
The next necessary file I will del:

**Comment by NashMiao** [comment]
The next unnecessary file I will delete:

image

File: swift/Podfile.lock (hunk 19 above)
Comment (hewigovens) [reply swift/Podfile.lock]: You should merge hsiaosiyuan0/wallet-core first, it's still old code
